packages/wallet-utils/src/lib/create-action.ts

```diff
@@ -0,0 +1,74 @@
+import * as BN from "bn.js";
+import { utils, transactions } from "near-api-js";
+import { AddKeyPermission, Action } from "@near-wallet-selector/core";
+
+const getAccessKey = (permission: AddKeyPermission) => {
+  if (permission === "FullAccess") {
+    return transactions.fullAccessKey();
+  }
+
+  const { receiverId, methodNames = [] } = permission;
+  const allowance = permission.allowance
+    ? new BN(permission.allowance)
+    : undefined;
+
+  return transactions.functionCallAccessKey(receiverId, methodNames, allowance);
+};
+
+export const createAction = (actions: Array<Action>) => {
```

Comment:
I think there's been some confusion on this. The type for this method should be:

```ts
createAction(action: Action): transactions.Action;
```

The idea is the consumer of this function will use `.map` if needed